### ISSUE TITLE
Choosing node version based on target directory

### ISSUE THREAD

**Issue by marcelklehr** [jasongin/nvs#23]
Something that might be a little unintuitive is that you have to cd into a directory where a version requirement is set before that becomes active.

E.g.
```
$ echo "v6.0.0" > .node-version
$ cd ..
$ node app/index.js
```

The node version might not be 6.0.0, even though the directory setting says it should be. I know this is an edge case, but something one should be aware of imo ;)

**Comment by jasongin** [comment]
Yes, the nvs auto-switching function is based on the current working directory in the shell, not the directory of a target script. I understand it might not be intuitive to everyone, but I think it fits in more naturally with the way nvs works in other ways by editing `PATH`.

The only way to support *implicitly* choosing a node version based on a target script location would be to *shim* the node executable. While certainly possible (as done by some other node version manager tools), I have avoided shimming as a switching mechanism in nvs, because it is mostly redundant with `PATH` changes, can cause problems with IDEs and debugging, and may not behave as expected with `child_process.fork()`.

If this kind of functionality is *explicitly* desired in some cases, what do you think about an extension of the `nvs auto` command that would work like this?
```sh
$ echo "v6.0.0" > .node-version
$ cd ..
$ nvs auto app/index.js
```
When a script path is passed to `nvs auto`, it will search for a `.node-version` file in the target script directory (or parent directories), and use that version to run the script. This would be an alternative to the existing `nvs run` command that runs a script using a node version specified on the command-line (instead of in a `.node-version` file).

**Comment by ljharb** [comment]
Generally, a project should use the same node version throughout, and all commands in a project should only ever be run from the project root. What's the use case for `app` to have a different node version than the project it's contained in?

**Comment by marcelklehr** [comment]
I guess it's not that big of a problem if you consider the project root rule as a given. My point was that it's not necessarily intuitive and that it might be worth the time to investigate edge cases for this design, which I haven't done, yet. So, I can't tell you specific cases that would require target dir inspection. I made the choice to use shims in my project, so I was interested in the considerations and the implications of this design.
Come to think about it, specifying a node version per target script dir *could* help resolve #21. But then it might not be worth the cost.
